unstable test TestVerifyCommonNameAndRotate

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
[2022-09-14T09:04:02.702Z] ==================== Test output for //util:util_test (shard 16 of 50):
[2022-09-14T09:04:02.702Z] === RUN   TestVerifyCommonNameAndRotate
[2022-09-14T09:04:02.703Z] 2022/09/14 17:03:59 http: TLS handshake error from 127.0.0.1:39084: client certificate authentication failed. The Common Name from the client certificate [client2 ] was not found in the configuration cluster-verify-cn with value: [client1]
[2022-09-14T09:04:02.703Z]     security_test.go:115: 
[2022-09-14T09:04:02.703Z]         	Error Trace:	/home/jenkins/.tidb/tmp/b7c8fa3cc70b7c7023fdc4f79d50df0e/sandbox/processwrapper-sandbox/3801/execroot/__main__/bazel-out/k8-fastbuild/bin/util/util_test_/util_test.runfiles/__main__/util/security_test.go:115
[2022-09-14T09:04:02.703Z]         	Error:      	Error "Get \"https://127.0.0.1:9292\": write tcp 127.0.0.1:39084->127.0.0.1:9292: write: broken pipe" does not contain "tls: bad certificate"
[2022-09-14T09:04:02.703Z]         	Test:       	TestVerifyCommonNameAndRotate
[2022-09-14T09:04:02.703Z] --- FAIL: TestVerifyCommonNameAndRotate (38.14s)
```
https://ci.pingcap.net/blue/rest/organizations/jenkins/pipelines/tidb_ghpr_unit_test/runs/69213/nodes/59/steps/64/log/?start=0
<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/pkg/util/security_test.go
+++ b/pkg/util/security_test.go
@@ -75,14 +75,13 @@ func TestVerifyCommonNameAndRotate(t *testing.T) {
 		util.WithVerifyCommonName([]string{"client1"}),
 	)
 	require.NoError(t, err)
-	port := 9292
-	url := fmt.Sprintf("https://127.0.0.1:%d", port)
 	ctx, cancel := context.WithCancel(context.Background())
-	server := runServer(ctx, serverTLS, port, t)
+	server, port := runServer(ctx, serverTLS, t)
 	defer func() {
 		cancel()
 		server.Close()
 	}()
+	url := fmt.Sprintf("https://127.0.0.1:%d", port)
 
 	clientTLS1, err := util.NewTLSConfig(
 		util.WithCAContent(caData),
@@ -143,14 +142,13 @@ func TestCA(t *testing.T) {
 		util.WithCertAndKeyContent(serverCert, serverKey),
 	)
 	require.NoError(t, err)
-	port := 9293
-	url := fmt.Sprintf("https://127.0.0.1:%d", port)
 	ctx, cancel := context.WithCancel(context.Background())
-	server := runServer(ctx, serverTLS, port, t)
+	server, port := runServer(ctx, serverTLS, t)
 	defer func() {
 		cancel()
 		server.Close()
 	}()
+	url := fmt.Sprintf("https://127.0.0.1:%d", port)
 
 	// test only CA
 	clientTLS1, err := util.NewTLSConfig(
@@ -202,18 +200,19 @@ func handler(w http.ResponseWriter, req *http.Request) {
 	w.Write([]byte("This an example server"))
 }
 
-func runServer(ctx context.Context, tlsCfg *tls.Config, port int, t *testing.T) *http.Server {
+func runServer(ctx context.Context, tlsCfg *tls.Config, t *testing.T) (*http.Server, int) {
 	http.HandleFunc("/", handler)
-	server := &http.Server{Addr: fmt.Sprintf(":%d", port), Handler: nil}
+	server := &http.Server{Addr: ":0", Handler: nil}
 
 	conn, err := net.Listen("tcp", server.Addr)
 	if err != nil {
 		require.NoError(t, err)
 	}
+	port := conn.Addr().(*net.TCPAddr).Port
 
 	tlsListener := tls.NewListener(conn, tlsCfg)
 	go server.Serve(tlsListener)
-	return server
+	return server, port
 }
 
 // generateCerts returns the PEM contents of a CA certificate and some certificates and private keys per Common Name in

